pkg/fleet/env/env.go

```diff
@@ -0,0 +1,131 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package env provides the environment variables for the installer.
+package env
+
+import (
+	"os"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/utils"
+)
+
+const (
+	envAPIKey         = "DD_API_KEY"
+	envSite           = "DD_SITE"
+	envRegistry       = "DD_INSTALLER_REGISTRY"
+	envRegistryAuth   = "DD_INSTALLER_REGISTRY_AUTH"
+	envDefaultVersion = "DD_INSTALLER_DEFAULT_VERSION"
+)
+
+var defaultEnv = Env{
+	APIKey: "",
+	Site:   "datadoghq.com",
+
+	RegistryOverride:                "",
+	RegistryAuthOverride:            "",
+	RegistryOverrideByImage:         map[string]string{},
+	DefaultVersionOverrideByPackage: map[string]string{},
+}
+
+// Env contains the configuration for the installer.
+type Env struct {
+	APIKey string
+	Site   string
+
+	RegistryOverride                string
+	RegistryAuthOverride            string
+	RegistryOverrideByImage         map[string]string
+	RegistryAuthOverrideByImage     map[string]string
+	DefaultVersionOverrideByPackage map[string]string
+}
+
+// FromEnv returns an Env struct with values from the environment.
+func FromEnv() *Env {
+	return &Env{
+		APIKey: getEnvOrDefault(envAPIKey, defaultEnv.APIKey),
+		Site:   getEnvOrDefault(envSite, defaultEnv.Site),
+
+		RegistryOverride:                getEnvOrDefault(envRegistry, defaultEnv.RegistryOverride),
+		RegistryAuthOverride:            getEnvOrDefault(envRegistryAuth, defaultEnv.RegistryAuthOverride),
+		RegistryOverrideByImage:         overridesByNameFromEnv(envRegistry, []string{envRegistryAuth}),
+		RegistryAuthOverrideByImage:     overridesByNameFromEnv(envRegistryAuth, []string{}),
+		DefaultVersionOverrideByPackage: overridesByNameFromEnv(envDefaultVersion, []string{}),
+	}
+}
+
+// FromConfig returns an Env struct with values from the configuration.
+func FromConfig(config config.Reader) *Env {
+	return &Env{
```

Comment:
nit: If you're reading from config too shouldn't we rename this package & the Env struct to Config or something?